s2spy/_base_calendar.py

```diff
@@ -156,10 +164,10 @@ def _set_year_range_from_timestamps(self):
 
         # ensure that the input data could always cover the advent calendar
         # last date check
-        if self._map_year(map_last_year).iloc[0].right > self._last_timestamp:
+        while self._map_year(map_last_year).iloc[0].right > self._last_timestamp:
```

Comment:
```suggestion
        while self._map_year(last_year).iloc[0].right > self._last_timestamp:
```
Would it make sense to rename this? I got a bit confused by the "double mapping".

Reply:
Looking at it, `min_year` and `max_year` might be a more appropriate name. I find `first` and `last` a bit ambiguous with the BackwardCalendar.